src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -175,418 +170,124 @@ public String handleSaveFile(MultipartFile file, boolean keepFileName, boolean m
+                + "." + fileExtension;
+        return folder.resolve(generatedFilename);
     }
 
     /**
-     * Takes care of any changes that have to be made to the filesystem (deleting old files, moving temporary files into their proper location) and returns the public path for the
-     * resulting file (as it might have been moved from newFilePath to another path)
+     * Copies an existing non-temporary file to a target location.
      *
-     * @param oldFilePath  the old file path (this file will be deleted if not null and different from newFilePath)
-     * @param newFilePath  the new file path (this file will be moved into its proper location, if it was a temporary file)
-     * @param targetFolder the folder that a temporary file should be moved to
-     * @param entityId     id of the entity this file belongs to (needed to generate public path). If this is null, a placeholder will be inserted where the id would be
-     * @param keepFileName flag for determining if the current filename should be kept.
-     * @return the resulting public path (is identical to newFilePath, if file didn't need to be moved)
+     * @param oldFilePath  the old file path
+     * @param targetFolder the folder that a file should be copied to
+     * @return the resulting file path or null on error
      */
-    public String manageFilesForUpdatedFilePath(String oldFilePath, String newFilePath, String targetFolder, Long entityId, Boolean keepFileName) {
-        if (oldFilePath != null) {
-            if (oldFilePath.equals(newFilePath)) {
-                // Do nothing
-                return newFilePath;
-            }
-            else {
-                // delete old file
-                log.debug("Delete old file {}", oldFilePath);
-                try {
-                    File oldFile = new File(actualPathForPublicPathOrThrow(oldFilePath));
-
-                    if (!FileSystemUtils.deleteRecursively(oldFile)) {
-                        log.warn("FileService.manageFilesForUpdatedFilePath: Could not delete old file: {}", oldFile);
-                    }
-                    else {
-                        log.debug("Deleted Orphaned File: {}", oldFile);
-                    }
-                }
-                catch (Exception ex) {
-                    log.warn("FileService.manageFilesForUpdatedFilePath: Could not delete old file '{}' due to exception {}", oldFilePath, ex.getMessage());
-                }
-            }
-        }
-
-        return moveFileIfTemporaryAndReturnPath(newFilePath, targetFolder, entityId, keepFileName);
-    }
-
-    private String moveFileIfTemporaryAndReturnPath(String path, String targetFolder, Long entityId, Boolean keepFileName) {
-        if (path != null && path.contains("files/temp")) {
-            // rename and move file
+    public Path copyExistingFileToTarget(Path oldFilePath, Path targetFolder) {
+        if (oldFilePath != null && !pathContains(oldFilePath, Path.of(("files/temp")))) {
```

Comment:
```suggestion
        if (oldFilePath == null || pathContains(oldFilePath, Path.of(("files/temp")))) {
            return null;
        }
```
Indentation level of the method body can be reduced by inverting the condition.